src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -336,4 +325,22 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
             return file;
         };
     }
+
+    /**
+     * Extract the bucket from a gs path.
+     */
+    public static String extractBucketFromPath(final String path) {
+        if (!path.startsWith("gs://") && StringUtils.countMatches(path, ',') < 3) {
+            throw new PdaoException(String.format("Bad GS path %s", path));
+        }
+        return path.substring(5).split("/")[0];
```

Comment:
Can you add a comment on what this is doing?

Reply:
You got it